### PR TITLE
cmake: don't do enable_language + add -x objective-c + check for OBJC

### PR DESCRIPTION
**Pull request by madebr**
Doing `enable_language` in SDL, when included as a subproject causes problems on CMake.
```
CMake Error: Error required internal CMake variable not set, cmake may not be built correctly.
Missing variable is:
CMAKE_OBJC_COMPILE_OBJECT
```
Avoid this by not doing `enable_language(OBJC` and making sure to add `-x objective-c` ourselves.

FIxes https://github.com/libsdl-org/SDL/pull/6198
Fixes https://github.com/libsdl-org/SDL/issues/6454

@pthom @dashandslash

### CODE REVIEW

**Comment by slouken** [comment]
We’ve gone back and forth over this several times over the years. Can you add some more documentation in the CMake file why you’re not using the built-in objective c support?

**Comment by pthom** [comment]
I tested this PR on my side and it does work when building SDL static or shared, with Xcode, Ninja or make generators.

This was on a Mac M1, using cmake 3.24.2, Apple clang version 13.1.6, MacOS (Monterey V12.6), XCode (Version 13.4.1 (13F100))

**Comment by madebr** [comment]
I've added these lines:
```
  # - CMake versions <3.16 do not support the OBJC language
  # - When SDL is built as a subproject and when the main project does not enable OBJC,
  #   CMake fails due to missing internal CMake variables (CMAKE_OBJC_COMPILE_OBJECT)
  #   (reproduced with CMake 2.24.2)
```

@pthom 
I've changed the pr to keep the current behavior when SDL is built as the main project.
Only when SDL is built as a subproject, CMake will fall back to the old behavior.
We might want to report this issue upstream. Let's discuss about a reproducer in https://github.com/libsdl-org/SDL/issues/6454.

**Comment by pthom** [comment]
This is a likely due to a bug in CMake: 

minimal reproduction code for the cmake bug here:

https://github.com/pthom/cmake_objc_issue

File: cmake/macros.cmake [hunk 1]
```diff
@@ -92,7 +92,11 @@ macro(LISTTOSTRREV _LIST _OUTPUT)
   endforeach()
 endmacro()
 
-if(${CMAKE_VERSION} VERSION_LESS "3.16.0")
+if(CMAKE_VERSION VERSION_LESS 3.16.0 OR SDL2_SUBPROJECT)
+  # - CMake versions <3.16 do not support the OBJC language
+  # - When SDL is built as a subproject and when the main project does not enable OBJC,
+  #   CMake fails due to missing internal CMake variables (CMAKE_OBJC_COMPILE_OBJECT)
+  #   (reproduced with CMake 2.24.2)
```
Comment (pthom) [inline cmake/macros.cmake:99]: CMake 3.24.2

Comment (madebr) [reply cmake/macros.cmake]: Doh!
Fixed in 6c63a96cca224b31913a98ff3e42e72cdb4c0f24
Thanks!

**Comment by genpfault** [comment]
Works For Me(TM) on a Macmini6,1 running macOS 10.15, Homebrew's cmake & ninja, and consuming SDL & friends via `add_subdirectory()`.

**Pull request merged**